stepsizers.go

```diff
@@ -115,3 +118,61 @@ func (q *QuadraticStepSize) StepSize(l Location, dir []float64) (stepSize float6
 	copy(q.xPrev, l.X)
 	return stepSize
 }
+
+// FirstOrderStepSize estimates the initial line search step size based on the
+// assumption that the first-order change in the function will be the same as
+// that obtained at the previous iteration. That is, the initial step size s^0_k
+// is chosen so that
+//   s^0_k ∇f_k⋅p_k = s_{k-1} ∇f_{k-1}⋅p_{k-1}
+// This is useful for line search methods that do not produce well-scaled
+// descent directions, such as gradient descent or conjugate gradient methods.
+type FirstOrderStepSize struct {
```

Comment:
Reading these, I think it would be nicer if the struct comments follow the Go style. "Initial step factor sets the step size for the first iteration to be |g|_inf / InitialStepFactor"

I know this isn't consistent at the moment, but reading these for something I don't know makes me realize I've gotten used to reading godoc style comments.